### ISSUE TITLE
Prevent from hitting Google Safe API limit

### ISSUE THREAD

**Issue by abhijithvijayan** [thedevs-network/kutt#169]
kutt.it redirects with 429

**Comment by webbzac** [comment]
Yeah same issue here, `{"error":"Request failed with status code 429"}`.

429 status code is for [Too Many Requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/429), so it's likely that there is some rate limiting going on for whatever reason.

**Comment by abhijithvijayan** [comment]
cant create links with the extension. previous links work.

**Comment by poeti8** [comment]
Yes, sorry it's for Google Safe API limit, seems like some spammers tried to abuse Kutt. I'm gonna find a way to prevent them from doing so in future.

**Comment by uhlhosting** [comment]
@poeti8 why not using some honeypot?

**Comment by Couto** [comment]
Just as an idea and I haven't checked the code to see if this is implemented, but Google Safe results could be cached in redis, this way duplicated urls would avoid hitting the API for a certain time (defined in the cache).

**Comment by trgwii** [comment]
@poeti8 do you know if the attacks were all with the same URL? If not we'd have to think of something smarter

**Comment by TheCapsLock** [comment]
Why not just hitting google at all ?

**Comment by abusayed0206** [comment]
Same erro fro custm domain still exists!
